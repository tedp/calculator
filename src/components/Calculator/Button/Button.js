import React from 'react';
import './Button.scss';

class Button extends React.Component {
  handleClick = () => {
    this.props.onClick(this.props.value);
  }

  render() {
    return (
      <div className="Button" onClick={this.handleClick}>
        {this.props.value}
      </div>
    );
  }
}

export default Button
