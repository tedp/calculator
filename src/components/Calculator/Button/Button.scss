@import "../../../styles/styles.scss";


$module: "Button";

.#{$module} {
  display: inline-block;
  border: 1px solid $mid-grey;
  border-radius: 5px;
  width: $button-width;
  text-align: center;
  padding: $padding;
  margin: 20px 4px 10px 0;
  cursor: pointer;
  background-color: $dark-grey;

  &:hover {
    background-color: $dark-grey;
    border-color: $light-grey;
    @include box-shadow();
  }

  &:active {
    font-weight: bold;
  }
}
