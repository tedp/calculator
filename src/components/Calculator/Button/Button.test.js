import React from 'react';
import { shallow } from 'enzyme';
import Button from './Button';


describe('Button', () => {
  const clickFn = jest.fn();

  it('should render correctly', () => {
    const component = shallow(<Button />);

    expect(component).toMatchSnapshot();
  });

  it('should render button correctly with given value', () => {
    const value = 'value';
    const component = shallow(<Button value={value} />);

    expect(component).toMatchSnapshot();
  });

  it('button click should provide button value', () => {
    const value = 'value';
    const component = shallow(<Button value={value} onClick={clickFn} />);
    component.simulate('click');

    expect(clickFn).toHaveBeenCalledWith(value);
  });
});
