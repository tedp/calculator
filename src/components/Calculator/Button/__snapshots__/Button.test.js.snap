// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render button correctly with given value 1`] = `
<div
  className="Button"
  onClick={[Function]}
>
  value
</div>
`;

exports[`Button should render correctly 1`] = `
<div
  className="Button"
  onClick={[Function]}
/>
`;
